dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -1352,4 +1357,47 @@ protected void setUserAPI(UserAPI userAPI) {
 	public boolean validateMe() {
 		return !UtilMethods.isSet(map.get(Contentlet.DONT_VALIDATE_ME));
 	}
+
+	/**
+	 * Returns a list of all contentlets related to this instance given a RelationshipField variable
+	 * @param variableName
+	 * @return
+	 */
+	public List<Contentlet> getRelated(final String variableName) {
+
+		if (this.relatedIds.containsKey(variableName)) {
+			return this.relatedIds.get(variableName).stream().map(identifier -> {
+				try {
+					return this.contentletAPI.findContentletByIdentifierAnyLanguage(identifier);
+				} catch (DotDataException | DotSecurityException e) {
+					throw new DotStateException(e);
+				}
+			}).collect(Collectors.toList());
```

Comment:
see if this could be immutable list (collectionUtils.immutableList()) would be useful